### ISSUE TITLE
zlib vulnerability CVE-2023-45853

### ISSUE THREAD

**Issue by michaelbilow** [oauth2-proxy/oauth2-proxy#2325]
## Expected Behavior

Update the zlib version. See #1862 and #1861.

## Current Behavior

There is a critical level vulnerability in a zlib dependency: https://security-tracker.debian.org/tracker/CVE-2023-45853

## Steps to Reproduce (for bugs)

```docker
FROM python:3.9.18-bookworm
ENV OAUTH2_PROXY_VERSION v7.5.1
RUN wget https://github.com/oauth2-proxy/oauth2-proxy/releases/download/${OAUTH2_PROXY_VERSION}/oauth2-proxy-${OAUTH2_PROXY_VER
SION}.linux-amd64.tar.gz -O /tmp/oauth2-proxy.tar.gz
RUN tar xvfz /tmp/oauth2-proxy.tar.gz
RUN mkdir /opt/oauth2-proxy
RUN mv oauth2-proxy-${OAUTH2_PROXY_VERSION}.linux-amd64/oauth2-proxy /usr/local/bin/oauth2-proxy
RUN chmod +x /usr/local/bin/oauth2-proxy
RUN oauth2-proxy --version
```

```bash
$ docker build . -t python-oauth2
$ trivy image python-oauth2
```

## Context

Complaints due to having a Critical CVE in a dependency.

## Your Environment

- Version used: 7.5.1
- OS: Debian Bookworm

**Comment by JoelSpeed** [comment]
We are looking at removing zlib from the current image and [switching to distroless](https://github.com/oauth2-proxy/oauth2-proxy/pull/2295), it isn't currently used by OAuth2 Proxy. While yes, it's in the image, it shouldn't be exploitable as part of the day to day running of the application.

**Comment by kvanzuijlen** [comment]
This vulnerability was fixed in v7.6.0
